src/Magento/FunctionalTestingFramework/Util/Path/UrlFormatter.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\Util\Path;
+
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+
+class UrlFormatter implements FileUrlFormatterInterface
+{
+    /**
+     * Return formatted url path from input string, or false on error
+     *
+     * @param string  $url
+     * @param boolean $withTrailingSeparator
+     * @return string
+     * @throws TestFrameworkException
+     */
+    public static function format($url, $withTrailingSeparator = true)
+    {
+        $sanitizedUrl = rtrim($url, '/');
+        $sanitizedUrl = filter_var($sanitizedUrl, FILTER_SANITIZE_URL);
+
+        if (false === $sanitizedUrl) {
```

Comment:
Please add some comments on each of the lines before the `if` statements (where applicable)
Something like `Check if URL is in format: <format>`

Reply:
Added